tests/components/tasmota/test_light.py

```diff
@@ -0,0 +1,747 @@
+"""The tests for the Tasmota light platform."""
+import copy
+import json
+
+from homeassistant.components import light
+from homeassistant.components.light import (
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_COLOR_TEMP,
+    SUPPORT_EFFECT,
+    SUPPORT_TRANSITION,
+    SUPPORT_WHITE_VALUE,
+)
+from homeassistant.components.tasmota.const import DEFAULT_PREFIX
+from homeassistant.const import ATTR_ASSUMED_STATE, STATE_OFF, STATE_ON
+
+from .test_common import (
+    DEFAULT_CONFIG,
+    help_test_availability,
+    help_test_availability_discovery_update,
+    help_test_availability_poll_state,
+    help_test_availability_when_connection_lost,
+    help_test_discovery_device_remove,
+    help_test_discovery_removal,
+    help_test_discovery_update_unchanged,
+    help_test_entity_id_update_discovery_update,
+    help_test_entity_id_update_subscriptions,
+)
+
+from tests.async_mock import patch
+from tests.common import async_fire_mqtt_message
+from tests.components.light import common
+
+
+async def test_attributes_on_off(hass, mqtt_mock, setup_tasmota):
+    """Test state update via MQTT."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["rl"][0] = 1
+    config["so"]["30"] = 1  # Enforce Home Assistant auto-discovery as light
+    mac = config["mac"]
+
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/LWT", "Online")
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/STATE", '{"POWER":"ON"}')
+
+    state = hass.states.get("light.test")
+    assert state.attributes.get("effect_list") is None
+    assert state.attributes.get("min_mireds") is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)